pkg/experiment/ingester/segment.go

```diff
@@ -334,21 +334,31 @@ func concatSegmentHead(f *headFlush, w *writerOffset, s *metadata.StringTable) (
 		Labels:          nil,
 	}
 
+	var allLabels []int32
 	lb := metadata.NewLabelBuilder(s)
 	for _, profileType := range ptypes {
 		lb.WithLabelSet(model.LabelNameServiceName, f.head.key.service, model.LabelNameProfileType, profileType)
+		labelSet := lb.Build() // This creates one complete label set with count prefix
+		allLabels = append(allLabels, labelSet...)
+	}
+
+	if f.head.needsSymbolization {
+		// lb.WithLabelSet(metadata.LabelNameNeedsSymbolization, "true")
+		lb.WithLabelSet(model.LabelNameServiceName, f.head.key.service, metadata.LabelNameNeedsSymbolization, "true")
+		labelSet := lb.Build()
+		allLabels = append(allLabels, labelSet...)
 	}
 
 	// Other optional labels:
 	// lb.WithLabelSet("label_name", "label_value", ...)
-	ds.Labels = lb.Build()
+	ds.Labels = allLabels
```

Comment:
Well, `Build` should be called once, after all the labels where added. If it does not work, it's a bug in the label builder that should be fixed

Reply:
It was a non-related issue to this. Changed it back :)